File: test/corpus/python/clean.py
# Cohesive: each function calls the next, so the file is one connected component
# and trips no metric. The true-negative case.

LO = 0.0
HI = 1.0


def clampv(x, lo, hi):
    if x < lo:
        return lo
    if x > hi:
        return hi
    return x


def normalize(x):
    return clampv(x, LO, HI)


def scale(x, k):
    return normalize(x) * k


def accumulate(xs):
    total = 0.0
    for x in xs:
        total += scale(x, 2)
    return total

File: test/corpus/go/clean.go
package corpus

// Cohesive: each function calls the next, so the file is one connected component.

const LO = 0.0
const HI = 1.0

func clampv(x float64, lo float64, hi float64) float64 {
    if x < lo {
        return lo
    }
    if x > hi {
        return hi
    }
    return x
}

func normalize(x float64) float64 {
    return clampv(x, LO, HI)
}

func scale(x float64, k float64) float64 {
    return normalize(x) * k
}

func accumulate(xs []float64) float64 {
    total := 0.0
    for i := 0; i < len(xs); i++ {
        total += scale(xs[i], 2)
    }
    return total
}
